lib/hawkular/operations/operations_api.rb

```diff
@@ -128,46 +128,113 @@ def invoke_specific_operation(operation_payload, operation_name, &callback)
       invoke_operation_helper(operation_payload, operation_name, &callback)
     end
 
-    # Deploys a war file into WildFly
+    # Deploys an archive file into WildFly
     #
     # @param [Hash] hash Arguments for deployment
     # @option hash [String]  :resource_path canonical path of the WildFly server into which we deploy
     # @option hash [String]  :destination_file_name resulting file name
     # @option hash [String]  :binary_content binary content representing the war file
-    # @option hash [String]  :enabled whether the deployment should be enabled or not
+    # @option hash [String]  :enabled whether the deployment should be enabled immediately, or not (default = true)
+    # @option hash [String]  :force_deploy whether to replace existing content or not (default = false)
+    # @option hash [String]  :server_groups comma-separated list of server groups for the operation (default = ignored)
     #
     # @param callback [Block] callback that is run after the operation is done
     def add_deployment(hash, &callback)
       hash[:enabled] = hash.key?(:enabled) ? hash[:enabled] : true
+      hash[:force_deploy] = hash.key?(:force_deploy) ? hash[:force_deploy] : false
       required = [:resource_path, :destination_file_name, :binary_content]
       check_pre_conditions hash, required, &callback
 
       operation_payload = prepare_payload_hash([:binary_content], hash)
       invoke_operation_helper(operation_payload, 'DeployApplication', hash[:binary_content], &callback)
     end
 
-    # Removes an existing deployment from WildFly
+    # Undeploy a WildFly deployment
     #
     # @param [Hash] hash Arguments for deployment removal
-    # @option hash [String]  :resource_path canonical path of the WildFly server from which to remove the deployment
-    # @option hash [String]  :deployment_name name of deployment to remove
+    # @option hash [String]  :resource_path canonical path of the WildFly server from which to undeploy the deployment
+    # @option hash [String]  :deployment_name name of deployment to undeploy
+    # @option hash [String]  :remove_content whether to remove the deployment content or not (default = true)
+    # @option hash [String]  :server_groups comma-separated list of server groups for the operation (default = ignored)
     #
     # @param callback [Block] callback that is run after the operation is done
-    def remove_deployment(hash, &callback)
+    def undeploy(hash, &callback)
```

Comment:
I'm not sure if there is already anyone using 'remove_deployment'.
If so, we could use [this](http://stackoverflow.com/a/4763178) to create an alias and mark it as deprecated.